packages/medusa-payment-stripe/src/core/stripe-base.ts

```diff
@@ -0,0 +1,327 @@
+      return intent as unknown as PaymentProcessorSessionResponse["session_data"]
+    } catch (error) {
+      if (error.code === ErrorCodes.PAYMENT_INTENT_UNEXPECTED_STATE) {
+        if (error.payment_intent?.status === ErrorIntentStatus.SUCCEEDED) {
+          return error.payment_intent
+        }
+      }
+
+      return this.buildError(
+        "An error occurred in deletePayment during the capture of the payment",
+        error
+      )
+    }
+  }
+
+  async deletePayment(
+    paymentSessionData: Record<string, unknown>
+  ): Promise<
+    PaymentProcessorError | PaymentProcessorSessionResponse["session_data"]
+  > {
+    return await this.cancelPayment(paymentSessionData)
+  }
+
+  async refundPayment(
+    paymentSessionData: Record<string, unknown>,
+    refundAmount: number
+  ): Promise<
+    PaymentProcessorError | PaymentProcessorSessionResponse["session_data"]
+  > {
+    const id = paymentSessionData.id as string
+
+    try {
+      await this.stripe_.refunds.create({
+        amount: Math.round(refundAmount),
+        payment_intent: id as string,
+      })
+    } catch (e) {
+      return this.buildError(
+        "An error occurred in refundPayment during the refundPayment",
+        e
+      )
+    }
+
+    return paymentSessionData
+  }
+
+  async retrievePayment(
+    paymentSessionData: Record<string, unknown>
+  ): Promise<
+    PaymentProcessorError | PaymentProcessorSessionResponse["session_data"]
+  > {
+    try {
+      const id = paymentSessionData.id as string
+      const intent = await this.stripe_.paymentIntents.retrieve(id)
+      return intent as unknown as PaymentProcessorSessionResponse["session_data"]
+    } catch (e) {
+      return this.buildError("An error occurred in retrievePayment", e)
+    }
+  }
+
+  async updatePayment(
+    context: PaymentProcessorContext
+  ): Promise<PaymentProcessorError | PaymentProcessorSessionResponse | void> {
+    const { amount, customer, paymentSessionData } = context
+    const stripeId = customer?.metadata?.stripe_id
+
+    if (stripeId !== paymentSessionData.customer) {
+      const result = await this.initiatePayment(context)
+      if (isPaymentProcessorError(result)) {
+        return this.buildError(
+          "An error occurred in updatePayment during the initiate of the new payment for the new customer",
+          result
+        )
+      }
+
+      return result
+    } else {
+      if (amount && paymentSessionData.amount === Math.round(amount)) {
+        return
+      }
+
+      try {
+        const id = paymentSessionData.id as string
+        const sessionData = (await this.stripe_.paymentIntents.update(id, {
+          amount: Math.round(amount),
+        })) as unknown as PaymentProcessorSessionResponse["session_data"]
+
+        return { session_data: sessionData }
+      } catch (e) {
+        return this.buildError(
+          "An error occurred in updatePayment during the update of the payment",
+          e
```

Comment:
todo: Simplify error message
```suggestion
          "An error occurred in updatePayment",
          e
```